### PR TITLE
add observer.version_major

### PR DESCRIPTION
**Pull request by graphaelli**
for identifiying which major version of apm-server created docs

relates to #1802

### CODE REVIEW

File: publish/pub.go [hunk 1]
```diff
@@ -70,11 +70,12 @@ func NewPublisher(info beat.Info, pipeline beat.Pipeline, shutdownTimeout time.D
...
 			"observer": common.MapStr{
-				"type":         info.Beat,
-				"hostname":     info.Hostname,
-				"version":      info.Version,
-				"id":           info.ID.String(),
-				"ephemeral_id": info.EphemeralID.String(),
+				"type":          info.Beat,
+				"hostname":      info.Hostname,
+				"version":       info.Version,
+				"version_major": 7,
```
Comment (jalvz) [inline publish/pub.go]: why this is hardcoded and not derived from info.Version?

Comment (simitt) [reply publish/pub.go]: I assumed it was for performance reasons.

Comment (jalvz) [reply publish/pub.go]: performance of `strings.Split("info.Version", ".")[0]`?
ok, was just curious. i guess this code won't be here by 8.0 :+1:

Comment (graphaelli) [reply publish/pub.go]: This was to avoid any chance of a runtime error.  A test was added so we don't forget to change this for 8.0.

**Pull request merged**